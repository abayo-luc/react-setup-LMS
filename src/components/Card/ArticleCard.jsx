import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import './style.scss';

const ArticleCard = ({ item }) => {
  const { title, description, updatedAt } = item;
  return (
    <div className="card mb-3" style={{ maxWidth: '100%' }}>
      <div className="row no-gutters">
        <div className="col-md-4">
          <img
            // eslint-disable-next-line max-len
            src="https://res.cloudinary.com/nexpals/image/upload/v1530951150/cover_z7omnk.jpg"
            alt="img"
            className="card-img"
          />
        </div>
        <div className="col-md-8">
          <div className="card-body">
            <h5 className="card-title">{title}</h5>
            <p className="card-text">{description}</p>
            <p className="card-text">
              <small className="text-muted">
                Last updated
                {' '}
                {moment(moment(updatedAt), 'YYYYMMDD').fromNow()}
              </small>
            </p>
          </div>
        </div>
      </div>
    </div>
  );
};
ArticleCard.propTypes = {
  item: PropTypes.shape({
    title: PropTypes.string,
    description: PropTypes.string,
    createdAt: PropTypes.string,
  }).isRequired,
};
export default ArticleCard;
