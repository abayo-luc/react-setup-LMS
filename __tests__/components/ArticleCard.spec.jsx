import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ArticleCard from '../../src/components/Card/ArticleCard';

const props = {
  item: {
    title: '',
    description: '',
    updatedAt: new Date(),
  },
};
const warper = shallow(<ArticleCard {...props} />);
describe('Article Card Component', () => {
  test('should match the snapshots', () => {
    expect(toJson(warper)).toMatchSnapshot();
  });
});
