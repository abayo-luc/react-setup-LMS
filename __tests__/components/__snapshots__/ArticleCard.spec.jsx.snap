// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Card Component should match the snapshots 1`] = `
<div
  className="card mb-3"
  style={
    Object {
      "maxWidth": "100%",
    }
  }
>
  <div
    className="row no-gutters"
  >
    <div
      className="col-md-4"
    >
      <img
        alt="img"
        className="card-img"
        src="https://res.cloudinary.com/nexpals/image/upload/v1530951150/cover_z7omnk.jpg"
      />
    </div>
    <div
      className="col-md-8"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        />
        <p
          className="card-text"
        />
        <p
          className="card-text"
        >
          <small
            className="text-muted"
          >
            Last updated
             
            a few seconds ago
          </small>
        </p>
      </div>
    </div>
  </div>
</div>
`;
